### ISSUE TITLE
Code Actions not working (tested on gitsigns)

### ISSUE THREAD

**Issue by c3n21** [jose-elias-alvarez/null-ls.nvim#1611]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

NVIM v0.10.0-dev-601+gd55d7646c

### Dev Version?

- [ ] I am using a stable Neovim release version, or if I am using a dev version of Neovim I have confirmed that my issue is reproducible on a stable version.

### Operating System

ArchLinux (WSL)

### Minimal Config

```Lua
-- this template is borrowed from nvim-lspconfig
local on_windows = vim.loop.os_uname().version:match("Windows")

local function join_paths(...)
	local path_sep = on_windows and "\\" or "/"
	local result = table.concat({ ... }, path_sep)
	return result
end

vim.g.loaded_remote_plugins = ""
vim.cmd([[set runtimepath=$VIMRUNTIME]])

local temp_dir = vim.loop.os_getenv("TEMP") or "/tmp"

vim.cmd("set packpath=" .. join_paths(temp_dir, "nvim", "site"))

local package_root = join_paths(temp_dir, "nvim", "site", "pack")
local install_path = join_paths(package_root, "packer", "start", "packer.nvim")
local compile_path = join_paths(install_path, "plugin", "packer_compiled.lua")

local null_ls_config = function()
	local null_ls = require("null-ls")
	-- add only what you need to reproduce your issue
	require("gitsigns").setup({
		signs = {
			add = { hl = "GitSignsAdd", text = "│", numhl = "GitSignsAddNr", linehl = "GitSignsAddLn" },
			change = {
				hl = "GitSignsChange",
				text = "│",
				numhl = "GitSignsChangeNr",
				linehl = "GitSignsChangeLn",
			},
			delete = {
				hl = "GitSignsDelete",
				text = "_",
				numhl = "GitSignsDeleteNr",
				linehl = "GitSignsDeleteLn",
			},
			topdelete = {
				hl = "GitSignsDelete",
				text = "‾",
				numhl = "GitSignsDeleteNr",
				linehl = "GitSignsDeleteLn",
			},
			changedelete = {
				hl = "GitSignsChange",
				text = "~",
				numhl = "GitSignsChangeNr",
				linehl = "GitSignsChangeLn",
			},
		},
		signcolumn = true, -- Toggle with `:Gitsigns toggle_signs`
		numhl = true, -- Toggle with `:Gitsigns toggle_numhl`
		linehl = true, -- Toggle with `:Gitsigns toggle_linehl`
		word_diff = true, -- Toggle with `:Gitsigns toggle_word_diff`
		watch_gitdir = {
			interval = 1000,
			follow_files = true,
		},
		attach_to_untracked = true,
		current_line_blame = true, -- Toggle with `:Gitsigns toggle_current_line_blame`
		current_line_blame_opts = {
			virt_text = true,
			virt_text_pos = "eol", -- 'eol' | 'overlay' | 'right_align'
			delay = 1000,
			ignore_whitespace = false,
		},
		current_line_blame_formatter = "<author>, <author_time:%Y-%m-%d> - <summary>",
		sign_priority = 6,
		update_debounce = 100,
		status_formatter = nil, -- Use default
		max_file_length = 40000,
		preview_config = {
			-- Options passed to nvim_open_win
			border = "single",
			style = "minimal",
			relative = "cursor",
			row = 0,
			col = 1,
		},
		yadm = {
			enable = false,
		},
		on_attach = function()
			local gs = require("gitsigns")

			vim.keymap.set("n", "<leader>hs", gs.stage_hunk)
			vim.keymap.set("n", "<leader>hr", gs.reset_hunk)
			vim.keymap.set("n", "<leader>hu", gs.undo_stage_hunk)

			vim.keymap.set("v", "<leader>hs", function()
				gs.stage_hunk({ vim.fn.line("."), vim.fn.line("v") })
			end)

			vim.keymap.set("v", "<leader>hr", function()
				gs.reset_hunk({ vim.fn.line("."), vim.fn.line("v") })
			end)

			vim.keymap.set("n", "<leader>hS", gs.stage_buffer)
			vim.keymap.set("n", "<leader>hR", gs.reset_buffer)
			vim.keymap.set("n", "<leader>hp", gs.preview_hunk)
			vim.keymap.set("n", "<leader>hb", function()
				gs.blame_line({ full = true })
			end)

			vim.keymap.set("n", "<leader>tb", gs.toggle_current_line_blame)
			vim.keymap.set("n", "<leader>hd", gs.diffthis)
			vim.keymap.set("n", "<leader>hD", function()
				gs.diffthis("~")
			end)
			vim.keymap.set("n", "<leader>td", gs.toggle_deleted)

			-- Text object
			vim.keymap.set({ "o", "x" }, "ih", ":<C-U>Gitsigns select_hunk<CR>")

			vim.keymap.set({ "n" }, "]c", function()
				if vim.wo.diff then
					return "]c"
				end
				vim.schedule(function()
					require("gitsigns").next_hunk()
				end)
				return "<Ignore>"
			end, { expr = true })

			vim.keymap.set({ "n" }, "[c", function()
				if vim.wo.diff then
					return "[c"
				end
				vim.schedule(function()
					require("gitsigns").prev_hunk()
				end)
				return "<Ignore>"
			end, { expr = true })
		end,
	})
	local null_ls = require("null-ls")
	local sources = {
		null_ls.builtins.code_actions.gitsigns,
	}

	null_ls.setup({
		sources = sources,
		debug = true,
	})

	vim.keymap.set({ "n", "v" }, "<leader>ca", vim.lsp.buf.code_action, map_opts)
end

local function load_plugins()
	-- only add other plugins if they are necessary to reproduce the issue
	require("packer").startup({
		{
			"wbthomason/packer.nvim",
			{
				"jose-elias-alvarez/null-ls.nvim",
				requires = { "nvim-lua/plenary.nvim", "lewis6991/gitsigns.nvim" },
				config = null_ls_config,
			},
		},
		config = {
			package_root = package_root,
			compile_path = compile_path,
		},
	})
end

if vim.fn.isdirectory(install_path) == 0 then
	vim.fn.system({ "git", "clone", "https://github.com/wbthomason/packer.nvim", install_path })
	load_plugins()
	require("packer").sync()
else
	load_plugins()
	require("packer").sync()
end
```

### Steps to Reproduce

1. Go into a git repo
2. Edit some files without staging and/or committing
3. Run `lua vim.lsp.buf.code_action()`
4. Try any code actions and you will notice it won't work

I noticed that this only happens on the latest nightly, right now I'm on `1106079` commit (using `bob` as NeoVim version manager) and seems to work somehow 

### Reproducibility Check

- [x] I confirm that my minimal config is based on the `minimal_init.lua` template and that my issue is reproducible by running `nvim --clean -u minimal_init.lua` and following the steps above.

### Expected Behavior

When selecting Preview or Stage hunk it should respectively show me the preview or stage the hunk

### Actual Behavior

When running the first time the code action it will show `Language server `null-ls` does not support command `NULL_LS_CODE_ACTION`. This command may require a client extension.` and subsequent code actions with gitsigns won't display this message but it won't work.

### Debug Log

```
[TRACE Fri 30 Jun 2023 10:21:25 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:102: received LSP request for method shutdown
[TRACE Fri 30 Jun 2023 10:21:25 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:127: received LSP notification for method exit
[TRACE Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/client.lua:97: starting null-ls client
[TRACE Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:102: received LSP request for method initialize
[DEBUG Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/client.lua:152: unable to notify client for method textDocument/didOpen (client not active): {
  textDocument = {
    uri = "file:///home/zhifan/Downloads/github/codeactions_bug/init.lua"
  }
}
[TRACE Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:127: received LSP notification for method initialized
[TRACE Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:127: received LSP notification for method textDocument/didOpen
[TRACE Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_DIAGNOSTICS_ON_OPEN
[DEBUG Fri 30 Jun 2023 10:21:30 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:24: no generators available
[TRACE Fri 30 Jun 2023 10:21:37 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:102: received LSP request for method textDocument/codeAction
[TRACE Fri 30 Jun 2023 10:21:37 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_CODE_ACTION
[TRACE Fri 30 Jun 2023 10:21:37 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/code-actions.lua:35: received code actions from generators
[TRACE Fri 30 Jun 2023 10:21:37 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/code-actions.lua:36: { {
    arguments = {
      title = "Reset hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Reset hunk"
  }, {
    arguments = {
      title = "Preview hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Preview hunk"
  }, {
    arguments = {
      title = "Stage hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Stage hunk"
  }, {
    arguments = {
      title = "Select hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Select hunk"
  } }
[TRACE Fri 30 Jun 2023 10:21:40 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:102: received LSP request for method textDocument/codeAction
[TRACE Fri 30 Jun 2023 10:21:40 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_CODE_ACTION
[TRACE Fri 30 Jun 2023 10:21:40 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/code-actions.lua:35: received code actions from generators
[TRACE Fri 30 Jun 2023 10:21:40 AM CEST] /tmp/nvim/site/pack/packer/start/null-ls.nvim/lua/null-ls/code-actions.lua:36: { {
    arguments = {
      title = "Reset hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Reset hunk"
  }, {
    arguments = {
      title = "Preview hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Preview hunk"
  }, {
    arguments = {
      title = "Stage hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Stage hunk"
  }, {
    arguments = {
      title = "Select hunk"
    },
    command = "NULL_LS_CODE_ACTION",
    title = "Select hunk"
  } }
```

### Help

Yes, but I don't know how to start. I would need guidance

### Implementation Help

An explanation about what is happening under the hood and perhaps the possible cause of this issue.

### Requirements

- [X] I have read and followed the instructions above and understand that my issue will be closed if I did not provide the required information.

**Comment by samuelabreu** [comment]
I encounter this error trying to use refactoring on my setup, i guess [commit](https://github.com/neovim/neovim/commit/64f2691a984a5b1e2958d5656a910054982a6f0e) changed a behavior which null-ls depends.

Using neovim 0.9.1 it works (kinda, on lunarvim it throws an error on first use, then works fine, didn't test without lunarvim)

**Comment by jose-elias-alvarez** [comment]
> I encounter this error trying to use refactoring on my setup, i guess [commit](https://github.com/neovim/neovim/commit/64f2691a984a5b1e2958d5656a910054982a6f0e) changed a behavior which null-ls depends.
> 
> Using neovim 0.9.1 it works (kinda, on lunarvim it throws an error on first use, then works fine, didn't test without lunarvim)

I can confirm that the minimal init works fine on 0.9.1 (not sure what's going on with LunarVim). 

Based on the description and the linked  commit, I think changing [this line](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/aac27a1fa550de3d0b2c651168167cc0d5366a9a/lua/null-ls/rpc.lua#L32) to return a table containing `methods.internal.CODE_ACTION` instead of just `true` should fix the issue. 

I don't have the bandwidth to support nightly versions myself, so a user will have to contribute the fix (and also make sure that the fix doesn't affect 0.9.1).

**Comment by samuelabreu** [comment]
I opened https://github.com/jose-elias-alvarez/null-ls.nvim/pull/1613, sorry for multiple commits, don't actually have my env setup for lua, if neede i can reopen with only 1 commit
